core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -1187,6 +1187,9 @@ private static RelDataType arrayAppendPrependReturnType(SqlOperatorBinding opBin
     if (elementType.isNullable()) {
       type = opBinding.getTypeFactory().createTypeWithNullability(type, true);
     }
+    SqlValidatorUtil.
+        adjustTypeForArrayFunctionConstructor(componentType, elementType, opBinding);
```

Comment:
why pass elementType? the leastRestrictive type is `type` here

Reply:
thank you